File: week3/Python/977. Squares of a Sorted Array .py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        result = [0]*(len(nums))
        l = 0
        r = len(nums)-1
        for i in range(len(nums)-1, -1, -1):
            if l == r:
                result[i] = nums[r]**2
            if nums[l]**2 > nums[r]**2:
                result[i]=nums[l]**2
                l += 1      
            else:
                result[i] = nums[r]**2      
                r -= 1          
        return result

File: week3/Java/977. Squares of a Sorted Array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int left = 0, right = nums.length-1;
        int index = nums.length-1;
        
        int[] result = new int[right+1];
        while(index >= 0) {
            if(Math.abs(nums[left]) > Math.abs(nums[right])) {
                result[index--] = nums[left]*nums[left];
                left++;
            } else {
                result[index--] = nums[right]*nums[right];
                right--;
            }
        }
        
        return result;
    }
}
